src/System.ComponentModel.Annotations/tests/RegularExpressionAttributeTests.cs

```diff
@@ -2,78 +2,57 @@
-            AssertEx.Empty(attribute.Pattern);
-            Assert.Throws<InvalidOperationException>(
-                () => attribute.Validate("Does not matter - Pattern is empty", s_testValidationContext));
+            var attribute = new RegularExpressionAttribute(pattern);
+            Assert.Equal(pattern, attribute.Pattern);
         }
 
-        [Fact]
-        public static void Validate_successful_for_null_or_empty_value()
+        [Theory]
+        [InlineData(12345)]
+        public static void MatchTimeoutInMilliseconds_GetSet_ReturnsExpected(int newValue)
         {
             var attribute = new RegularExpressionAttribute("SomePattern");
-            AssertEx.DoesNotThrow(() => attribute.Validate(null, s_testValidationContext)); // Null is valid
-            AssertEx.DoesNotThrow(() => attribute.Validate(string.Empty, s_testValidationContext)); // Empty string is valid
+            attribute.MatchTimeoutInMilliseconds = newValue;
+            Assert.Equal(newValue, attribute.MatchTimeoutInMilliseconds);
         }
 
-        [Fact]
-        public static void Validate_successful_for_value_matching_pattern()
+        [Theory]
+        [InlineData(null)]
+        [InlineData("")]
+        public static void Validate_InvalidPattern_ThrowsInvalidOperationException(string pattern)
         {
-            var attribute = new RegularExpressionAttribute("defghi");
-            attribute.MatchTimeoutInMilliseconds = 5000; // note: timeout is just a number much larger than we expect the test to take
-            AssertEx.DoesNotThrow(() => attribute.Validate("defghi", s_testValidationContext));
-
-            attribute = new RegularExpressionAttribute("[^a]+\\.[^z]+");
-            attribute.MatchTimeoutInMilliseconds = 10000; // note: timeout is just a number much larger than we expect the test to take
-            AssertEx.DoesNotThrow(() => attribute.Validate("bcdefghijklmnopqrstuvwxyz.abcdefghijklmnopqrstuvwxy", s_testValidationContext));
-        }
-
-        [Fact]
-        public static void Validate_throws_for_values_which_do_not_match_pattern()
-        {
-            var attribute = new RegularExpressionAttribute("defghi");
-            Assert.Throws<ValidationException>(() => attribute.Validate("zyxwvu", s_testValidationContext)); // pattern does not match
-            Assert.Throws<ValidationException>(() => attribute.Validate("defghijkl", s_testValidationContext)); // pattern only matches beginning of value
-            Assert.Throws<ValidationException>(() => attribute.Validate("abcdefghi", s_testValidationContext)); // pattern only matches end of value
-            Assert.Throws<ValidationException>(() => attribute.Validate("abcdefghijkl", s_testValidationContext)); // pattern only matches part of value
-
-            attribute = new RegularExpressionAttribute("[^a]+\\.[^z]+");
-            attribute.MatchTimeoutInMilliseconds = 10000; // note: timeout is just a number much larger than we expect the test to take
```

Comment:
I added a test for a timeout of zero (throws ArgumentOutOfRangeException, which is probably a bug!)
However, I can't figure out how to make the regex evaluation to timeout - even something like this; 
```

yield return new TestCase(new RegularExpressionAttribute(new string('a', 50000)) { MatchTimeoutInMilliseconds = 1 }, new string('a', 50000));
```
passes!